File: src/development/docker.py
import os
import subprocess

# Set up the repository
os.system('sudo dnf -y install dnf-plugins-core')
os.system('sudo dnf config-manager --add-repo https://download.docker.com/linux/fedora/docker-ce.repo')

# Install Docker Engine
os.system('sudo dnf install docker-ce docker-ce-cli containerd.io docker-compose-plugin')
os.system('sudo systemctl start docker')

# Enable Docker Engine
os.system('sudo systemctl enable docker.service')
os.system('sudo systemctl enable containerd.service')

# Docker Engine post-installation steps
os.system('echo -e "\nYou can manage Docker as a non-root user. For more information, please visit https://docs.docker.com/engine/install/linux-postinstall"')

File: src/scripts/docker.rs
use anyhow::Result;
use cmd_lib::run_cmd;
use colored::Colorize;

pub fn install_docker() -> Result<()> {
    println!("{}", "Installing Docker...".green().bold());

    let user = std::env::var("USER")?;

    // Add Docker repository
    let repo_url = "https://download.docker.com/linux/fedora/docker-ce.repo";
    run_cmd!(sudo dnf config-manager addrepo --from-repofile=$repo_url --overwrite)?;

    run_cmd!(sudo dnf update -y)?;

    // Install Docker
    run_cmd!(sudo dnf install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin -y)?;

    // Start and enable Docker
    run_cmd!(sudo systemctl enable --now docker)?;

    // Add user to docker group
    run_cmd!(sudo usermod -aG docker $user)?;

    println!("{}", "✓ Docker installed successfully!".green().bold());
    println!(
        "\n{} Please log out and log back in so that your group membership is re-evaluated.",
        "⚠".yellow().bold()
    );

    Ok(())
}
